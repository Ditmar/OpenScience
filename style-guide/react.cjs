const { resolve } = require('node:path');
const project = resolve(process.cwd(), 'tsconfig.json');

module.exports = {
  env: {
    browser: true,
    es2021: true,
    es6: true,
    jest: true,
  },
  extends: [
    require.resolve('./rules/base'),
    require.resolve('./configurations/_base'),
    'plugin:react-hooks/recommended',
    'airbnb',
    'plugin:storybook/recommended',
    require.resolve('./configurations/typescript'),
    require.resolve('./configurations/vitest'),
  ],
  parserOptions: {
    project,
    babelOptions: {
      presets: ['@babel/preset-react'],
    },
  },
  plugins: ['react-refresh'],
  settings: {
    react: {
      version: 'detect',
    },
  },
  globals: {
    JSX: true,
  },
  rules: {
    'react/react-in-jsx-scope': 'off',
    'react/button-has-type': 'off',
    'react-refresh/only-export-components': [
      'warn',
      { allowExportNames: ['meta', 'links', 'headers', 'loader', 'action'] },
    ],
  },
};
