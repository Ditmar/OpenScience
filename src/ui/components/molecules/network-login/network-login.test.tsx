import React from 'react';
import { render } from '@testing-library/react';
import NetworkLogin from './network-login';

describe('NetworkLogin component', () => {
  it('renders correctly', () => {
    const { container } = render(<NetworkLogin />);
    expect(container).toMatchSnapshot();
  });
});
