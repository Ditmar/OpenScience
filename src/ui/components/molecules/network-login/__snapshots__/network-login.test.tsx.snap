// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NetworkLogin component > renders correctly 1`] = `
<div>
  <div
    class="panel panel--primary"
    data-testid="net-login"
  >
    <div
      class="logo logo--primary"
    >
      <span
        class="logo-icon"
        data-testid="logo_icon"
      >
        <!--?xml version="1.0" standalone="no"?-->
        <svg
          preserveAspectRatio="xMidYMid meet"
          version="1.0"
          viewBox="0 0 578.000000 288.000000"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="#000000"
            stroke="none"
            transform="translate(0.000000,288.000000) scale(0.100000,-0.100000)"
          >
            <path
              d="M2351 2476 c-8 -9 -11 -85 -9 -257 l3 -244 180 -5 180 -5 0 -25 c0
-24 -3 -25 -80 -32 -44 -4 -91 -10 -105 -13 -24 -6 -24 -7 -5 -18 23 -14 452
-17 500 -4 47 13 11 26 -94 34 l-91 6 0 28 c0 24 4 29 25 29 23 0 25 4 25 45
l0 45 -240 0 -240 0 0 195 0 195 370 0 370 0 0 -50 c0 -47 2 -50 25 -50 26 0
26 1 23 68 l-3 67 -411 3 c-341 2 -413 0 -423 -12z m419 -452 c0 -9 -5 -12
-12 -8 -8 5 -9 1 -4 -12 6 -15 5 -16 -5 -7 -10 10 -9 15 1 28 16 19 20 19 20
-1z m20 -14 c0 -11 -4 -20 -9 -20 -5 0 -7 9 -4 20 3 11 7 20 9 20 2 0 4 -9 4
-20z"
            />
            <path
              d="M2915 2327 c-3 -7 -4 -91 -3 -187 l3 -175 73 -3 c59 -2 72 0 72 13 0
11 -11 15 -40 15 l-40 0 0 155 0 155 215 0 215 0 0 -155 0 -155 -35 0 c-25 0
-35 -4 -35 -15 0 -13 12 -15 68 -13 l67 3 0 185 0 185 -278 3 c-220 2 -279 0
-282 -11z m553 -180 c2 -12 -3 -17 -17 -17 -12 0 -21 6 -21 13 0 31 32 34 38
4z"
            />
            <path
              d="M3087 2234 c-4 -4 -7 -90 -7 -191 l0 -183 115 0 115 0 -2 188 -3 187
-106 3 c-58 1 -108 -1 -112 -4z m126 -31 c-7 -2 -21 -2 -30 0 -10 3 -4 5 12 5
17 0 24 -2 18 -5z m77 -148 l0 -125 -95 0 -95 0 0 125 0 125 95 0 95 0 0 -125z
m-80 -171 c0 -16 -16 -19 -25 -4 -8 13 4 32 16 25 5 -4 9 -13 9 -21z"
            />
            <path
              d="M4662 1417 c-42 -34 -48 -43 -33 -45 17 -3 131 71 131 84 0 2 -10 4
-22 4 -13 0 -47 -19 -76 -43z"
            />
            <path
              d="M480 1035 c0 -278 0 -285 20 -285 20 0 20 7 20 285 0 278 0 285 -20
285 -20 0 -20 -7 -20 -285z"
            />
            <path
              d="M720 1035 c0 -278 0 -285 20 -285 19 0 20 7 20 246 0 204 2 245 13
235 8 -6 97 -116 198 -246 115 -147 191 -235 202 -235 16 0 17 20 17 285 0
277 -1 285 -20 285 -19 0 -20 -8 -22 -247 l-3 -247 -193 247 c-134 172 -199
247 -213 247 -18 0 -19 -9 -19 -285z"
            />
            <path
              d="M1514 1299 c-65 -25 -120 -78 -152 -147 -21 -47 -24 -65 -20 -131 7
-122 60 -200 168 -250 89 -41 209 -32 298 21 l32 20 0 109 c0 102 -1 109 -20
109 -18 0 -20 -8 -22 -102 l-3 -103 -40 -20 c-50 -25 -171 -25 -227 -1 -83 36
-148 134 -148 225 0 99 62 201 142 234 84 35 195 28 262 -18 28 -19 37 -21 45
-11 14 17 -9 38 -69 65 -60 28 -175 28 -246 0z"
            />
            <path
              d="M2030 1300 c0 -20 5 -20 182 -18 165 3 183 5 186 21 3 16 -11 17
-182 17 -180 0 -186 -1 -186 -20z"
            />
            <path
              d="M2580 1035 c0 -277 1 -285 20 -285 19 0 20 8 22 245 l3 245 191 -242
c104 -134 198 -244 207 -246 16 -3 17 19 17 282 0 278 -1 286 -20 286 -19 0
-20 -8 -22 -244 l-3 -245 -190 244 c-117 151 -197 245 -207 245 -17 0 -18 -20
-18 -285z"
            />
            <path
              d="M3250 1035 c0 -278 0 -285 20 -285 20 0 20 7 20 285 0 278 0 285 -20
285 -20 0 -20 -7 -20 -285z"
            />
            <path
              d="M3490 1300 c0 -19 7 -20 185 -20 178 0 185 1 185 20 0 19 -7 20 -185
20 -178 0 -185 -1 -185 -20z"
            />
            <path
              d="M4040 1301 c0 -20 5 -21 135 -21 148 0 182 -8 225 -54 34 -37 40
-137 11 -183 -36 -57 -69 -67 -228 -71 l-143 -4 0 -109 c0 -102 1 -109 20
-109 19 0 20 7 20 95 l0 96 112 -3 112 -3 65 -92 c51 -71 71 -93 89 -93 22 0
21 4 -24 68 -26 37 -58 81 -71 99 -24 31 -24 31 -3 38 30 10 79 53 96 87 21
41 20 130 -3 176 -11 23 -36 48 -67 67 -48 29 -53 30 -198 33 -145 4 -148 3
-148 -17z"
            />
            <path
              d="M4630 1035 c0 -278 0 -285 20 -285 20 0 20 7 20 285 0 278 0 285 -20
285 -20 0 -20 -7 -20 -285z"
            />
            <path
              d="M5018 1273 c-12 -27 -68 -149 -125 -273 -58 -124 -107 -231 -110
-238 -4 -10 1 -13 18 -10 20 3 39 38 139 255 63 139 117 254 120 257 3 2 57
-111 121 -252 98 -216 120 -258 139 -260 28 -5 37 -28 -113 300 -108 237 -125
268 -145 268 -18 0 -28 -10 -44 -47z"
            />
            <path
              d="M2030 1040 c0 -19 7 -20 185 -20 178 0 185 1 185 20 0 19 -7 20 -185
20 -178 0 -185 -1 -185 -20z"
            />
            <path
              d="M3490 1040 c0 -19 7 -20 185 -20 178 0 185 1 185 20 0 19 -7 20 -185
20 -178 0 -185 -1 -185 -20z"
            />
            <path
              d="M2030 765 c0 -13 27 -15 185 -15 158 0 185 2 185 15 0 13 -27 15
-185 15 -158 0 -185 -2 -185 -15z"
            />
            <path
              d="M3490 765 c0 -13 27 -15 185 -15 158 0 185 2 185 15 0 13 -27 15
-185 15 -158 0 -185 -2 -185 -15z"
            />
            <path
              d="M1710 650 c0 -6 32 -10 76 -10 74 0 77 -1 111 -35 33 -33 35 -39 30
-81 -8 -72 -39 -97 -127 -102 l-70 -4 0 56 c0 32 -4 56 -10 56 -7 0 -10 -24
-8 -62 3 -72 8 -75 106 -63 77 10 125 53 126 117 2 91 -55 138 -168 138 -37 0
-66 -4 -66 -10z"
            />
            <path
              d="M2020 650 c0 -6 33 -10 80 -10 47 0 80 4 80 10 0 6 -33 10 -80 10
-47 0 -80 -4 -80 -10z"
            />
            <path
              d="M2372 650 c-12 -5 -27 -21 -33 -34 -18 -38 9 -72 72 -90 63 -19 79
-32 79 -62 0 -42 -76 -59 -132 -30 -16 8 -28 10 -28 5 0 -18 72 -41 108 -35
102 16 93 108 -13 135 -70 17 -93 52 -58 86 14 14 29 16 77 12 56 -5 59 -4 36
8 -28 16 -74 18 -108 5z"
            />
            <path
              d="M2570 530 c0 -80 4 -130 10 -130 6 0 10 50 10 130 0 80 -4 130 -10
130 -6 0 -10 -50 -10 -130z"
            />
            <path
              d="M2702 650 c-12 -5 -27 -21 -32 -36 -16 -40 9 -70 71 -88 63 -19 79
-32 79 -62 0 -42 -76 -59 -132 -30 -16 8 -28 10 -28 5 0 -18 72 -41 108 -35
102 16 93 108 -13 135 -71 18 -93 52 -55 86 16 14 30 17 77 12 54 -5 56 -5 33
8 -28 16 -74 18 -108 5z"
            />
            <path
              d="M2860 650 c0 -5 20 -10 45 -10 l45 0 0 -121 c0 -77 4 -118 10 -114 6
4 10 55 10 121 l0 114 45 0 c25 0 45 5 45 10 0 6 -40 10 -100 10 -60 0 -100
-4 -100 -10z"
            />
            <path
              d="M3110 650 c0 -6 33 -10 80 -10 47 0 80 4 80 10 0 6 -33 10 -80 10
-47 0 -80 -4 -80 -10z"
            />
            <path
              d="M3350 530 c0 -159 18 -177 22 -23 l3 108 50 -88 c28 -48 52 -87 55
-87 3 0 28 39 55 86 l50 86 3 -98 c4 -153 22 -141 22 14 0 73 -3 132 -7 132
-5 0 -32 -43 -62 -95 -29 -52 -57 -95 -61 -95 -4 0 -32 43 -62 95 -29 52 -56
95 -60 95 -5 0 -8 -58 -8 -130z"
            />
            <path
              d="M3720 539 c-67 -148 -64 -139 -51 -139 6 0 11 4 11 9 0 5 23 58 51
117 l51 109 59 -120 c33 -66 57 -109 54 -95 -6 35 -103 240 -112 240 -5 0 -33
-55 -63 -121z"
            />
            <path
              d="M3990 653 c-52 -19 -68 -79 -27 -105 12 -8 44 -20 70 -28 59 -16 75
-41 52 -77 -15 -22 -22 -24 -63 -20 -26 2 -58 7 -72 11 -14 4 -7 -3 16 -15 70
-37 144 -10 144 51 0 33 -25 54 -75 66 -47 10 -75 32 -75 58 0 37 23 49 86 44
50 -4 55 -3 39 9 -19 13 -67 16 -95 6z"
            />
            <path
              d="M2053 533 c26 -2 67 -2 90 0 23 2 2 3 -48 3 -49 0 -68 -1 -42 -3z"
            />
            <path
              d="M3148 533 c23 -2 61 -2 85 0 23 2 4 4 -43 4 -47 0 -66 -2 -42 -4z"
            />
            <path
              d="M2015 410 c3 -5 39 -10 80 -10 41 0 77 5 80 10 4 6 -25 10 -80 10
-55 0 -84 -4 -80 -10z"
            />
            <path
              d="M3110 410 c0 -6 33 -10 80 -10 47 0 80 4 80 10 0 6 -33 10 -80 10
-47 0 -80 -4 -80 -10z"
            />
          </g>
        </svg>
      </span>
    </div>
    <div
      class="button-google"
    >
      <button
        class="button__label"
      >
        <div
          class="button button--undefined"
        >
          <span
            class="button__icon"
          >
            <svg
              fill="none"
              viewBox="0 0 21 21"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                clip-path="url(#clip0_3450_191)"
              >
                <path
                  d="M15.2645 8.1949C16.4119 8.1949 17.5615 8.23253 18.7072 8.18508C20.1354 8.12618 21.031 9.21864 20.9998 10.5004C20.8916 14.9531 17.998 18.9821 13.7887 20.4427C9.05383 22.0855 3.66 19.9845 1.28872 15.5738C-1.77645 9.87313 0.836321 2.8924 6.92077 0.662213C10.1744 -0.530058 13.2954 -0.0730024 16.2332 1.71868C16.8632 2.10319 16.9315 2.90277 16.4042 3.42582C15.8289 3.99577 15.2454 4.557 14.6651 5.12259C14.1859 5.58947 13.7559 5.66037 13.1566 5.35057C12.2332 4.87334 11.2585 4.63718 10.2143 4.66117C7.49174 4.7239 5.07566 6.81719 4.64074 9.50552C4.1042 12.8205 6.27714 15.841 9.6095 16.2539C12.0398 16.555 13.9903 15.6261 15.3721 13.5814C15.5814 13.2721 15.4186 12.9089 15.0498 12.8342C14.9531 12.8145 14.8509 12.8162 14.7509 12.8162C13.6855 12.8151 12.6195 12.8162 11.5541 12.8151C10.8454 12.814 10.4941 12.4644 10.4941 11.763C10.4941 10.9083 10.4924 10.0542 10.4941 9.19955C10.4952 8.5696 10.8602 8.19926 11.4929 8.19708C12.7495 8.19272 14.0067 8.19599 15.2639 8.19599L15.2645 8.1949Z"
                  fill="white"
                />
              </g>
              <defs>
                <clippath
                  id="clip0_3450_191"
                >
                  <rect
                    fill="white"
                    height="21"
                    width="21"
                  />
                </clippath>
              </defs>
            </svg>
          </span>
        </div>
        <span
          class="button__text"
        >
          Google
        </span>
      </button>
    </div>
    <div
      class="button-facebook"
    >
      <button
        class="button__label"
      >
        <div
          class="button button--undefined"
        >
          <span
            class="button__icon"
          >
            <svg
              fill="none"
              viewBox="0 0 15 33"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                clip-path="url(#clip0_3450_192)"
              >
                <path
                  d="M9.91179 33.0001V20.1801C9.91179 19.5743 9.91179 18.9693 9.91179 18.3635C9.91179 17.7534 9.90845 17.1434 9.91429 16.5103H14.4315C14.52 16.0994 15.0309 10.8776 14.9983 10.6989C14.8163 10.6989 14.6302 10.6989 14.4448 10.6989H11.0396C10.8526 10.6989 10.6656 10.7032 10.4786 10.6981C10.2958 10.693 10.1096 10.7236 9.92097 10.6742C9.91763 10.6028 9.91179 10.5424 9.91179 10.4819C9.91179 9.4073 9.91095 8.3318 9.91262 7.25716C9.91262 7.13549 9.92181 7.01211 9.9385 6.89129C10.0253 6.24122 10.4235 5.81494 11.0547 5.69922C11.159 5.6805 11.2675 5.67965 11.3736 5.67965C12.4822 5.67795 13.59 5.67965 14.6986 5.67965C14.7763 5.67965 14.8547 5.67965 14.9274 5.67965C14.9716 5.41503 14.9599 0.180503 14.9174 0.0247949C14.8681 0.0222423 14.8155 0.0188388 14.7637 0.0179879C14.4766 0.0162862 14.1894 0.0154353 13.9022 0.0145845C12.6876 0.0103301 11.4721 0.00947927 10.2574 0.00182148C9.54196 -0.00243286 8.82986 0.0299001 8.12695 0.178802C7.0133 0.415343 6.00985 0.876512 5.17336 1.67803C4.40116 2.41828 3.89192 3.32445 3.56968 4.34549C3.31256 5.16148 3.18984 5.99958 3.18566 6.8547C3.17982 8.05187 3.18399 9.24989 3.18399 10.4471C3.18399 10.5253 3.17565 10.6045 3.17064 10.7023H0V16.5146H3.18316V33.001H9.91179V33.0001Z"
                  fill="white"
                />
                <path
                  d="M9.91179 32.9999H3.18316V16.5135H0V10.7012H3.16897C3.17398 10.6034 3.18232 10.5251 3.18232 10.446C3.18232 9.24878 3.17815 8.05077 3.18399 6.8536C3.18817 5.99848 3.31089 5.15952 3.56801 4.34439C3.89025 3.3225 4.39949 2.41633 5.17169 1.67693C6.00818 0.87541 7.01247 0.41424 8.12528 0.177699C8.82903 0.0279468 9.54029 -0.00438612 10.2566 -0.00013179C11.4712 0.00752601 12.6867 0.00837687 13.9014 0.0126312C14.1886 0.0134821 14.4757 0.0143329 14.7629 0.0160347C14.8155 0.0160347 14.8673 0.020289 14.9165 0.0228416C14.9599 0.177699 14.9708 5.41308 14.9265 5.6777C14.8531 5.6777 14.7754 5.6777 14.6978 5.6777C13.5892 5.6777 12.4814 5.6777 11.3727 5.6777C11.2659 5.6777 11.1582 5.67855 11.0538 5.69727C10.4235 5.81214 10.0245 6.23927 9.93767 6.88933C9.92181 7.01016 9.91179 7.13353 9.91179 7.2552C9.91012 8.32985 9.91012 9.40534 9.91095 10.48C9.91095 10.5404 9.9168 10.6008 9.92014 10.6723C10.1088 10.7208 10.295 10.6902 10.4778 10.6961C10.6648 10.7021 10.8518 10.697 11.0388 10.697H14.444C14.6302 10.697 14.8155 10.697 14.9975 10.697C15.0301 10.8756 14.5191 16.0974 14.4307 16.5084H9.91346C9.90761 17.1423 9.91095 17.7523 9.91095 18.3616C9.91095 18.9674 9.91095 19.5723 9.91095 20.1782V32.9982L9.91179 32.9999Z"
                  fill="white"
                />
              </g>
              <defs>
                <clippath
                  id="clip0_3450_192"
                >
                  <rect
                    fill="white"
                    height="33"
                    width="15"
                  />
                </clippath>
              </defs>
            </svg>
          </span>
        </div>
        <span
          class="button__text"
        >
          Facebook
        </span>
      </button>
    </div>
  </div>
</div>
`;
